web_x2many_delete_all/static/src/js/web_x2many_delete_all.js

```diff
@@ -0,0 +1,37 @@
+/* Copyright 2016 Onestein
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_x2many_delete_all.Mixin', function (require) {
+    "use strict";
+    var core = require('web.core');
+    var _t = core._t;
+    var ListRenderer = require('web.ListRenderer');
+    
+    ListRenderer.include({
+        events: _.extend({}, ListRenderer.prototype.events, {
+            'click thead .o_list_record_delete_all': '_onTrashAllIconClick',
+        }),
+        _renderHeader: function (isGrouped) {
+            var $header = this._super.apply(this, arguments);
+            if (this.addTrashIcon) {
+                $header.find('tr').append(this._renderDeleteAll('th'));
+            }
+            return $header;
+        },
+        _renderDeleteAll: function (tag) {
+            var $icon = $('<button>', {class: 'fa fa-trash-o o_list_record_delete_all_btn', name: 'delete all',
```

Comment:
Maybe you can add btn btn-primary / btn-default and pull-right class to the button to make it look better.

Reply:
May be not. It looks ugly by applying btn-primary  or btn-danger. I have used the same properties which is there in delete button for line.